packages/graphql-dynamodb-transformer/src/resources.ts

```diff
@@ -252,13 +253,13 @@ export class ResourceFactory {
       RoleName: Fn.If(
         ResourceConstants.CONDITIONS.HasEnvironmentParameter,
         Fn.Join('-', [
-          typeName.slice(0, 21), // max of 64. 64-10-26-4-3 = 21
```

Comment:
I would still use the first 10-15 chars of the typeName in the role name. Or maybe even join the first 7 and last 7 and then concatenate on a slice of the hash at the end. With just the hash, the name has no meaning which makes it difficult to map back to what this role is actually for.